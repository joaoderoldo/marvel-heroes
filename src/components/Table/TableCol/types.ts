export interface TableColProps {
  children: React.ReactNode;
  classes?: string | string[];
  dataLabel?: string;
}
