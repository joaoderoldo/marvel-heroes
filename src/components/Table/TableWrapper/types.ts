import { ReactNode } from "react";

export interface TableWrapperProps {
  children: ReactNode;
  classes?: string | string[];
}
