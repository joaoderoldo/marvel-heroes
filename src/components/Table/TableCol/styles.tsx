import styled from "styled-components";

export const TableColWrapper = styled.div`
  width: calc(100% / 3);
  display: flex;
  align-items: center;

  &:last-child {
    justify-content: flex-end;
  }

  @media all and (max-width: 767px) {
    display: flex;
    padding: 10px 0;
    width: 100%;

    &:last-child {
      justify-content: flex-start;
    }

    &:before {
      color: #6c7a89;
      padding-right: 10px;
      content: attr(data-label);
      width: 40%;
      display: flex;
      flex-shrink: 0;
    }
  }
`;
