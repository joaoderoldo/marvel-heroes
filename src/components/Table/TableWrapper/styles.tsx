import styled from "styled-components";

export const TableWrapperWrapper = styled.div`
  margin-top: 32px;
`;
