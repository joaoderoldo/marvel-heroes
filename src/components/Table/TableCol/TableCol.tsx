import clsx from "clsx";
import { FC } from "react";

import { TableColWrapper } from "./styles";
import { TableColProps } from "./types";

const TableCol: FC<TableColProps> = ({ children, classes, dataLabel }) => {
  return (
    <TableColWrapper className={clsx(classes)} data-label={dataLabel}>
      {children}
    </TableColWrapper>
  );
};

export default TableCol;
