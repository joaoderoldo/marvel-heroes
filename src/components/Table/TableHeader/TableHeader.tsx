import { TableHeaderWrapper } from "./styles";

const TableHeader = ({ children, ...props }) => {
  return <TableHeaderWrapper {...props}>{children}</TableHeaderWrapper>;
};

export default TableHeader;
