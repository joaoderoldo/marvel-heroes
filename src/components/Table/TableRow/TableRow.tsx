import clsx from "clsx";
import { FC } from "react";

import { TableRowWrapper } from "./styles";
import { TableRowProps } from "./types";

const TableRow: FC<TableRowProps> = ({
  children,
  classes,
  onClick,
  clickable,
}) => {
  return (
    <TableRowWrapper className={clsx(classes)} onClick={onClick}>
      {children}
    </TableRowWrapper>
  );
};

export default TableRow;
