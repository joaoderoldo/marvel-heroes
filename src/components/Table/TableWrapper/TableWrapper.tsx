import { TableWrapperWrapper } from "./styles";

const TableWrapper = ({ children, ...props }) => {
  return <TableWrapperWrapper {...props}>{children}</TableWrapperWrapper>;
};

export { TableWrapper };
