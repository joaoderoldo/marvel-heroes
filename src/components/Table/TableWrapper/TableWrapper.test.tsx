import { render } from "@testing-library/react";
import { TableWrapper } from "./TableWrapper";
import TableHeader from "../TableHeader/TableHeader";
import TableRow from "../TableRow/TableRow";
import TableCol from "../TableCol/TableCol";

const TableMock = () => (
  <TableWrapper classes="table-wrapper">
    <TableHeader classes="table-header">
      <TableCol>Personagem</TableCol>
      <TableCol>SÃ©ries</TableCol>
      <TableCol>Eventos</TableCol>
    </TableHeader>
    <TableRow classes="table-row">
      <TableCol>Personagem Teste</TableCol>
      <TableCol>Serie Teste</TableCol>
      <TableCol>Evento Teste</TableCol>
    </TableRow>
  </TableWrapper>
);

describe("TableWrapper", () => {
  it("renders component correctly", () => {
    const { container } = render(<TableMock />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the children correctly", () => {
    const { getByText } = render(<TableMock />);
    expect(getByText("Personagem")).toBeInTheDocument();
  });

  it("renders with the correct className", () => {
    const { container } = render(<TableMock />);
    expect(container.firstChild).toHaveClass("table-wrapper");
  });
});
