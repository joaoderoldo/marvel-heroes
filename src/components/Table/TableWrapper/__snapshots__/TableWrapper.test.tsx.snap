// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableWrapper > renders component correctly 1`] = `
<div
  class="sc-beqWaB jlBTvX table-wrapper"
>
  <div
    class="sc-gueYoa kOOxlT table-header"
  >
    <div
      class="sc-hLseeU dFXXYa"
    >
      Personagem
    </div>
    <div
      class="sc-hLseeU dFXXYa"
    >
      SÃ©ries
    </div>
    <div
      class="sc-hLseeU dFXXYa"
    >
      Eventos
    </div>
  </div>
  <div
    class="sc-dmqHEX jqRwhR table-row"
  >
    <div
      class="sc-hLseeU dFXXYa"
    >
      Personagem Teste
    </div>
    <div
      class="sc-hLseeU dFXXYa"
    >
      Serie Teste
    </div>
    <div
      class="sc-hLseeU dFXXYa"
    >
      Evento Teste
    </div>
  </div>
</div>
`;
